src/game_api/rpc.cpp

```diff
@@ -806,7 +806,11 @@ void modify_sparktraps(float angle_increment, float distance)
     static float* new_array;
     if (new_array == nullptr)
     {
-        const auto offset = get_address("sparktrap_angle_increment");
+        const auto offset = get_address("sparktrap_angle_increment") + 4;
+
+        if (read_u8(offset - 1) != 0x89) // check if sparktraps_hack is active
+            return;
+
         const int32_t distance_offset = 0xF1;
         new_array = (float*)alloc_mem_rel32(offset + 4, sizeof(float) * 2);

```

Comment:
Also this offset is used here with `+4` again. What's the deal?

Reply:
the offset for jump/access memory is distance from past the expression, `alloc_mem_rel32` looks for the available memory above the given offset, so i have to correct for the 4 bits of the actual offset of the expression (otherwise if it would get free memory at exactly `offset - 32bit` it would be more then signed 32bit value cousin overflow)

hope that this explanation makes sense :stuck_out_tongue_closed_eyes:

Reply:
Doesn't really clear up why the other code that uses offset does not need to be modified for me. Was this code previously wrong?

Reply:
Ah yes, the missing piece for me was that I didn't notice that it was a `find_after_inst`, now it all makes sense.